flintrock/core.py

```diff
@@ -506,42 +506,66 @@ def get_java_major_version(client: paramiko.client.SSHClient):
                 client=client,
                 command=command)
             tokens = output.split()
-            # First line of the output is like: 'java version "1.8.0_20"'
+            # First line of the output is like: 'openjdk version "1.8.0_252"' or 'openjdk version "11.0.7" 2020-04-14'
             # Get the version string and strip out the first two parts of the
             # version as a tuple: (1, 8)
             if len(tokens) >= 3:
                 version_parts = tokens[2].strip('"').split(".")
                 if len(version_parts) >= 2:
-                    return tuple(int(part) for part in version_parts[:2])
+                    if version_parts[0] == "1":
+                        # Java 6, 7 or 8
+                        return version_parts[1]
+                    else:
+                        # Java 9+
+                        return version_parts[0]
         except SSHError:
             pass
 
     return None
 
 
-def ensure_java8(client: paramiko.client.SSHClient):
+def ensure_java(client: paramiko.client.SSHClient, jdk: str):
     host = client.get_transport().getpeername()[0]
     java_major_version = get_java_major_version(client)
 
-    if not java_major_version or java_major_version < (1, 8):
-        logger.info("[{h}] Installing Java 1.8...".format(h=host))
+    # sudo yum install -y java-1.8.0-openjdk
+    # sudo amazon-linux-extras install -y java-openjdk11
+    if not java_major_version or java_major_version < jdk:
```

Comment:
What should we do if the host has a newer version of Java than the one the user requested? At the very least, we should perhaps print some kind of warning. Ideally, we should ensure that the requested version of Java is activated even if a newer version is already present.